### PR TITLE
Convert API to promises

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "async": "~0.9.0",
     "babel-runtime": "^5.5.4",
     "bignumber.js": "^2.0.3",
+    "es6-promisify": "^2.0.0",
```
Comment (sublimator) [inline package.json]: I guess this will play nicely with yield and friends :) ?

**Comment by wltsmrz** [comment]
LGTM

File: package.json (hunk 1 above)
Comment (clark800) [inline package.json]: I'm not sure, is this what you mean? http://stackoverflow.com/questions/21846390/getting-a-promises-value-via-yield-co

File: src/api/common/schemas/settings-transaction.json [hunk 2]
```diff
@@ -10,6 +10,6 @@
     "address": {"$ref": "address"},
     "sequence": {"$ref": "sequence"}
   },
-  "required": ["type", "id", "address", "sequence", "specification", "outcome"],
+  "required": ["type", "id", "address", "sequence", "specification"],
```
Comment (lumberj) [inline src/api/common/schemas/settings-transaction.json]: Why was outcome removed?

Comment (clark800) [inline src/api/common/schemas/settings-transaction.json]: For non-validated transacitons.

Comment (lumberj) [inline src/api/common/schemas/settings-transaction.json]: I see, so you won't show an outcome on an unvalidated ledger?

Comment (clark800) [inline src/api/common/schemas/settings-transaction.json]: Yeah

File: src/api/ledger/accountinfo.js [hunk 3]
```diff
@@ -31,4 +31,8 @@ function getAccountInfo(account, options, callback) {
     composeAsync(formatAccountInfo, callback));
 }
 
-module.exports = utils.wrapCatch(getAccountInfo);
+function getAccountInfo(account: string, options={}) {
+  return utils.promisify(getAccountInfoAsync.bind(this))(account, options);
```
Comment (lumberj) [inline src/api/ledger/accountinfo.js]: What's the value of `this` here?

Comment (clark800) [inline src/api/ledger/accountinfo.js]: The instance of `RippleAPI` that this function gets attached to.

**Pull request merged**